src/coreclr/src/vm/callingconvention.h

```diff
@@ -1832,6 +1836,35 @@ class ArgIterator : public ArgIteratorTemplate<ArgIteratorBase>
         return FALSE;
 #endif
     }
+
+    BOOL HasValueTypeReturn()
+    {
+        WRAPPER_NO_CONTRACT;
+
+        TypeHandle thValueType;
+        CorElementType type = m_pSig->GetReturnTypeNormalized(&thValueType);
+        return type == ELEMENT_TYPE_VALUETYPE && !thValueType.IsEnum();
+    }
+
+#if defined(TARGET_X86) && !defined(FEATURE_STUBS_AS_IL)
+    BOOL HasMultiRegReturnStruct()
+    {
+        WRAPPER_NO_CONTRACT;
+
+        TypeHandle thValueType;
```

Comment:
This branch appears to be `HasValueTypeReturn()` - if that is indeed the check please use the helper function.